src/lib/conseil.js

```diff
@@ -0,0 +1,1101 @@
+    mintOperationQuery,
+    256
+  ) // TODO: this is hardwired and will not work for highly productive artists
+
+  const mintOperationResult = await conseiljs.TezosConseilClient.getTezosEntityData(
+    { url: conseilServer, apiKey: conseilApiKey, network: 'mainnet' },
+    'mainnet',
+    'operations',
+    mintOperationQuery
+  )
+
+  const operationGroupIds = mintOperationResult.map(
+    (r) => r['operation_group_hash']
+  )
+  const queryChunks = chunkArray(operationGroupIds, 30)
+
+  const makeObjectQuery = (opIds) => {
+    let mintedObjectsQuery = conseiljs.ConseilQueryBuilder.blankQuery()
+    mintedObjectsQuery = conseiljs.ConseilQueryBuilder.addFields(
+      mintedObjectsQuery,
+      'key_hash',
+      'value'
+    )
+    mintedObjectsQuery = conseiljs.ConseilQueryBuilder.addPredicate(
+      mintedObjectsQuery,
+      'big_map_id',
+      conseiljs.ConseilOperator.EQ,
+      [mainnet.nftMetadataMap]
+    )
+    mintedObjectsQuery = conseiljs.ConseilQueryBuilder.addPredicate(
+      mintedObjectsQuery,
+      'operation_group_id',
+      opIds.length > 1
+        ? conseiljs.ConseilOperator.IN
+        : conseiljs.ConseilOperator.EQ,
+      opIds
+    )
+    mintedObjectsQuery = conseiljs.ConseilQueryBuilder.setLimit(
+      mintedObjectsQuery,
+      opIds.length
+    )
+
+    return mintedObjectsQuery
+  }
+
+  const objectQueries = queryChunks.map((c) => makeObjectQuery(c))
+
+  const objectInfo = await Promise.all(
+    objectQueries.map(
+      async (q) =>
+        await conseiljs.TezosConseilClient.getTezosEntityData(
+          { url: conseilServer, apiKey: conseilApiKey, network: 'mainnet' },
+          'mainnet',
+          'big_map_contents',
+          q
+        ).then((result) =>
+          result.map((row) => {
+            const objectId = row['value']
+              .toString()
+              .replace(/^Pair ([0-9]{1,}) .*/, '$1')
+            const objectUrl = row['value']
+              .toString()
+              .replace(/.* 0x([0-9a-z]{1,}) \}$/, '$1')
+            const ipfsHash = Buffer.from(objectUrl, 'hex').toString().slice(7)
+
+            return { key: row['key_hash'], objectId, ipfsHash }
+          })
+        )
+    )
+  )
+
+  return objectInfo
+    .flat(1)
+    .sort((a, b) => parseInt(b.objectId) - parseInt(a.objectId))
+}
+
+const getArtisticUniverse = async (max_time) => {
+  var d = new Date()
+  d.setDate(d.getDate() - 14)
+  let mintOperationQuery = conseiljs.ConseilQueryBuilder.blankQuery()
+  mintOperationQuery = conseiljs.ConseilQueryBuilder.addFields(
+    mintOperationQuery,
+    'operation_group_hash'
+  )
+  mintOperationQuery = conseiljs.ConseilQueryBuilder.addPredicate(
+    mintOperationQuery,
+    'kind',
+    conseiljs.ConseilOperator.EQ,
+    ['transaction']
+  )
+  mintOperationQuery = conseiljs.ConseilQueryBuilder.addPredicate(
+    mintOperationQuery,
+    'timestamp',
+    conseiljs.ConseilOperator.BETWEEN,
+    [d.getTime(), max_time]
```

Comment:
/random seems to not work as it passes in zero. This seems to be an invalid range. No results returned locally.

Reply:
Seems like putting the earlier date first works (go figure). Not sure what the desired start date is as master is putting max_time (0 for /random) first.